source/cellular_bg96_api.c

```diff
@@ -3195,6 +3338,187 @@ CellularError_t Cellular_GetHostByName( CellularHandle_t cellularHandle,
+{
+    CellularContext_t* pContext = (CellularContext_t*)cellularHandle;
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+    uint32_t sendTimeout = DATA_SEND_TIMEOUT_MS;
+    char cmdBuf[CELLULAR_AT_CMD_TYPICAL_MAX_SIZE] = {'\0'};
+
+    CellularAtReq_t atReqUploadFile = {
+        cmdBuf,
+        CELLULAR_AT_NO_RESULT,
+        NULL,
+        NULL,
+        NULL,
+        0,
+    };
+
+    CellularAtDataReq_t atDataUploadFile =
+    {
+        fileContent,
+        fileSize,
+        pSentDataLength,
+        NULL,
+        0
+    };
+
+    if (cellularStatus == CELLULAR_SUCCESS)
+    {
+        /* Make sure the library is open. */
+        cellularStatus = _Cellular_CheckLibraryStatus(pContext);
+    }
+
+    if (cellularStatus == CELLULAR_SUCCESS)
+    {
+        /* Form the AT command. */
+
+        /* The return value of snprintf is not used.
+         * The max length of the string is fixed and checked offline. */
+        /* coverity[misra_c_2012_rule_21_6_violation]. */
+        (void)snprintf(cmdBuf, CELLULAR_AT_CMD_TYPICAL_MAX_SIZE, "%s%s,%d", "AT+QFUPL=", filename, fileSize);
+
+        pktStatus = _Cellular_AtcmdDataSend( pContext, atReqUploadFile, atDataUploadFile,
+                                             fileUploadInFileSystem, NULL,
+                                             PACKET_REQ_TIMEOUT_MS, sendTimeout, 0U );
+          
+        if (pktStatus != CELLULAR_PKT_STATUS_OK)
+        {
+            LogError(("Cellular_UploadFileToStorage: can't upload file: %s, cmdBuf:%s, PktRet: %d", filename, cmdBuf, pktStatus));
+            cellularStatus = _Cellular_TranslatePktStatus(pktStatus);
+        }
+    }
+
+    return cellularStatus;
+}
+
+CellularError_t Cellular_GetSpaceInformationOfTheStorage(CellularHandle_t cellularHandle, const char* storage_pattern)
+{
+    CellularContext_t* pContext = (CellularContext_t*)cellularHandle;
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+    char cmdBuf[CELLULAR_AT_CMD_TYPICAL_MAX_SIZE] = {'\0'};
+
+    CellularAtReq_t atReqGetSpaceInfo = {
+        cmdBuf,
+        CELLULAR_AT_NO_RESULT,
+        NULL,
+        NULL,
+        NULL,
+        0,
+    };
+
+    if (cellularStatus == CELLULAR_SUCCESS)
+    {
+        /* Make sure the library is open. */
+        cellularStatus = _Cellular_CheckLibraryStatus(pContext);
+    }
+
+    if (cellularStatus == CELLULAR_SUCCESS)
+    {
+        /* Form the AT command. */
+
+        /* The return value of snprintf is not used.
+         * The max length of the string is fixed and checked offline. */
+        /* coverity[misra_c_2012_rule_21_6_violation]. */
+        (void)snprintf(cmdBuf, CELLULAR_AT_CMD_TYPICAL_MAX_SIZE, "%s%s", "AT+QFLDS=", storage_pattern);
+        pktStatus = _Cellular_TimeoutAtcmdRequestWithCallback(pContext, atReqGetSpaceInfo, REQ_TIMEOUT_MS);

```

Comment:
QFLDS outputs

+QFLDS: <freesize>,<total_size>

But we're not doing anything with that output... ?

Reply:
Removed